test/test-tcp-try-write.c

```diff
@@ -43,12 +43,46 @@ static void close_cb(uv_handle_t* handle) {
 }
 
 
+static void bad_write_cb(uv_write_t *req, int status) {
+  ASSERT(0); // should be unreachable
+}
+
+
+static void bad_shutdown_cb(uv_shutdown_t *req, int status) {
+  ASSERT(0); // should be unreachable
+}
+
+
+static void bad_alloc_cb(uv_handle_t* handle, size_t size, uv_buf_t* buf) {
+  ASSERT(0); // should be unreachable
+}
+
+
+static void bad_read_cb(uv_stream_t* stream, ssize_t nread, const uv_buf_t* buf) {
+  ASSERT(0); // should be unreachable
+}
+
+
 static void connect_cb(uv_connect_t* req, int status) {
+  static uv_write_t write_req;
+  static uv_shutdown_t shutdown_req;
   int r;
   uv_buf_t buf;
   ASSERT(status == 0);
   connect_cb_called++;
 
+  buf = uv_buf_init("PING", 4);
+  r = uv_write(&write_req, (uv_stream_t*) &server, &buf, 1, bad_write_cb);
+  ASSERT(r == UV_EPIPE);
+  r = uv_shutdown(&shutdown_req, (uv_stream_t*) &server, bad_shutdown_cb);
+#ifdef _WIN32
+  ASSERT(r == UV_EPIPE);
+#else
+  ASSERT(r == -ENOTCONN);
```

Comment:
`UV_ENOTCONN`.  That said, divergence is not good.  Why not return `UV_EPIPE` on UNIX too?

Reply:
I'm just the messenger. The code that returns this error code already existed (but untested).